File: Course-Registration-System-main/java-(iteration_1)/src/CourseRequest.java
import java.util.ArrayList;

// A request for a course section by a student
public class CourseRequest {
    private int studentID;
    private CourseSection courseSectionCode;
    private String status;

    

    // Constructor the objects
    public CourseRequest(int studentID, CourseSection courseSection, String status) {
        this.studentID = studentID;
        this.courseSectionCode = courseSection;
        this.status = status;
    }
    // Retrieving student ID
    public int getStudentID() {
        return studentID;
    }

    // Setting student ID
    public void setStudentID(int studentID) {
        this.studentID = studentID;
    }

    // Retrieving course section
    public CourseSection getCourseSection() {
        return courseSectionCode;
    }

    // Setting course section
    public void setCourseSection(CourseSection courseSection) {
        this.courseSectionCode = courseSection;
    }

    // Retrieving status
    public String getStatus() {
        return status;
    }

    // Setting status
    public void setStatus(String status) {
        this.status = status;
    }

    // Displaying request info
    public String  viewRequestInfo() {
        return ("Student ID: " + getStudentID() + "\n" + getCourseSection().viewCourseSectionInfo() +
                "\nStatus: " + getStatus() + "\n-------------");
    }
}

File: Course-Registration-System-main/python-(iteration_2)/src/CourseRequest.py
class CourseRequest:
    """  A request for a course section by a student """


    """ Constructor for the objects """
    def __init__(self, studentID, courseSection, status):
        
        self.__studentID = studentID
        self.__courseSectionCode = courseSection
        self.__status = status
        
    def getCourseSectionCode(self):
        return self.__courseSectionCode.getCourseSectionCode()

    """ Retrieving student ID """
    def getStudentId(self):  
        return self.__studentID
    
    """ Setting student ID """
    def setStudentId(self, studentID):
        self.__studentID = studentID

    """ Retrieving course section """
    def getCourseSection(self):
        return self.__courseSectionCode
    
    """ Setting course section """
    def setCourseSection(self, courseSection):
        self.__courseSectionCode = courseSection

    """ Retrieving status """
    def getStatus(self):
        return self.__status
    
    """ Setting status """
    def setStatus(self, status):
        self.__status = status
    

    """ Displaying request info """
    def viewRequestInfo(self):
        
        return (
            f"Student ID: {self.getStudentId()}\n"
            f"{self.getCourseSection().viewCourseSectionInfo()}\n"
            f"Status: {self.getStatus()}\n-------------"
        )
